addon/lib/dependencies.js

```diff
@@ -0,0 +1,134 @@
+import { assert } from "@ember/debug";
+import { computed, get } from "@ember/object";
+
+import { getAST, getTransforms } from "ember-caluma/utils/jexl";
+
+/**
+ * Extract dependency slugs from a jexl expression.
+ *
+ * E.g: 'foo'|answer in 'bar'|answer|mapby('column') would result in the
+ * following dependency slugs: ['foo', 'bar', 'bar.column']
+ *
+ * @param {Jexl} jexl
+ * @param {String} expression
+ * @return {String[]} Returns an array of dependency slugs
+ */
+export function getDependenciesFromJexl(jexl, expression) {
+  const allTransforms = getTransforms(getAST(jexl, expression));
+  const answerTransforms = allTransforms.filter(
+    (transform) => transform.name === "answer"
+  );
+  const mapbyTransforms = allTransforms.filter(
+    (transform) =>
+      transform.name === "mapby" &&
+      transform.args[0].type === "FunctionCall" &&
+      transform.args[0].name === "answer"
+  );
```

Comment:
I tried following this using the JEXL playground: [see here](https://czosel.github.io/jexl-playground/#/?input='bar'%7Canswer%7Cmapby('column')&transforms=%7B%0A%20%20mapby%3A%20(name%2C%20col)%20%3D%3E%20name%20%2B%20%22col%3A%22%20%2Bcol%2C%0A%20%20answer%3A%20name%20%3D%3E%20%22Answer%20%22%20%2B%20name%2C%0A%7D).

However it seems that the AST structure looks a little different: The "mapby" only has the literal string as argument, and the part in front needs to be accessed via `subject`. Could it be that this changed recently in JEXL, or am I missing something else?

Input
```
'bar'|answer|mapby('column')
```

AST
```
{
  "type": "Transform",
  "name": "mapby",
  "args": [
    {
      "type": "Literal",
      "value": "column"
    }
  ],
  "subject": {
    "type": "Transform",
    "name": "answer",
    "args": [],
    "subject": {
      "type": "Literal",
      "value": "bar"
    }
  }
}
```

Reply:
Indeed, the playground needs an update. The AST in 2.3.0 is

```
{
  "type": "FunctionCall",
  "name": "mapby",
  "args": [
    {
      "type": "FunctionCall",
      "name": "answer",
      "args": [
        {
          "type": "Literal",
          "value": "bar"
        }
      ],
      "pool": "transforms"
    },
    {
      "type": "Literal",
      "value": "column"
    }
  ],
  "pool": "transforms"
}
```

Reply:
JEXL Playground updated :sparkles: